lib/gems/pending/appliance_console/database_configuration.rb

```diff
@@ -120,17 +120,22 @@ def ask_for_database_credentials
       self.username = just_ask("username", username) unless local?
       count = 0
       loop do
-        count += 1
-        password1   = ask_for_password_or_none("database password on #{host}", password)
+        password1 = ask_for_password("database password on #{host}", password)
         # if they took the default, just bail
         break if (password1 == password)
-        password2   = ask_for_password("database password again")
+        
+        if password1.strip.length == 0
+          say("\nPassword can not be empty, please try again")
+          next
+        end
+        password2 = ask_for_password("database password again")
         if password1 == password2
           self.password = password1
           break
-        elsif count > 1 # only reprompt password once
-          raise ArgumentError, "passwords did not match"
+        elsif count > 0 # only reprompt password once
```

Comment:
I wanted to preserve original logic: raise error if passwords do not match 2 times. 
If first attempt to enter password is empty than above logic would change and error will be raised on first attempt to enter not equal passwords.